nikola/plugins/task/galleries.py

```diff
@@ -589,9 +656,22 @@ def url_from_path(p):
                 },
                 'width': w,
                 'height': h
-            })
+            }
+        photo_array = []
+        if self.kw['sort_by_meta_file'] and context['order']:
+            for entry in context['order']:
+                photo_array.append(pre_photo.pop(entry))
+            # Do we have any orphan entries from metadata.yml?
+            if len(pre_photo) > 0:
+                for entry in pre_photo:
```

Comment:
I would sort entries here and below (in `else`) so the ordering does not change due to dict randomness (on Python ≤3.5)

Reply:
I'm uncertain about this suggestion.

We cannot sort `photo_info` since it's a dict, and unless we've got an order (implicitly or explicitly) specified from `metadata.yml`, then having unsorted output matches the existing behaviour. I'd prefer to leave this as-is.